src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -118,7 +169,7 @@ private function joinRelations(QueryBuilder $queryBuilder, string $resourceClass
 
             $relationAlias = $relationAlias.++$j;
 
-            $this->joinRelations($queryBuilder, $mapping['targetEntity'], $propertyMetadataOptions, $associationAlias, $relationAlias, $method === 'leftJoin');
+            $this->joinRelations($queryBuilder, $mapping['targetEntity'], $propertyMetadataOptions, $associationAlias, $relationAlias, $method === 'leftJoin', ++$depth);
```

Comment:
it is not, depends on what we mean my "depth", but as you said it's kinda the *level* of joined relations. 

Recursively each association may have more associations, each one counting +1. I put this there on my fork to throw when a mistake was made (like 2 associations joining each other will reach the max depth pretty fast).